testsuite/tests/kuadrant/authorino/dinosaur/conftest.py

```diff
@@ -86,49 +78,155 @@ def _resource_info(org_id, owner):
+    path_fourth_element = 'context.request.http.path.@extract:{"sep":"/","pos":4}'
+    path_third_element = 'context.request.http.path.@extract:{"sep":"/","pos":3}'
+    authorization.add_patterns(
         {
-            "NAME": blame("ac"),
-            "NAMESPACE": openshift.project,
-            "LABEL": module_label,
-            "HOST": hostname.hostname,
-            "RHSSO_ISSUER": rhsso.well_known["issuer"],
-            "ADMIN_ISSUER": admin_rhsso.well_known["issuer"],
-            "TERMS_AND_CONDITIONS": terms_and_conditions("false"),
-            "CLUSTER_INFO": cluster_info(rhsso.client_name),
-            "RESOURCE_INFO": resource_info("123", rhsso.client_name),
-        },
-    )
-    with openshift.context:
-        auth = selector.object(cls=AuthConfig)
-        request.addfinalizer(auth.delete)
-    route.add_auth_config(auth)
-    return auth
+            "api-route": [Pattern("context.request.http.path", "matches", "^/anything/dinosaurs_mgmt/.+")],
+            "v1-route": [Pattern(path_third_element, "eq", "v1")],
+            "dinosaurs-route": [Pattern(path_fourth_element, "eq", "dinosaurs")],
+            "dinosaur-resource-route": [Pattern("context.request.http.path", "matches", "/dinosaurs/[^/]+$")],
+            "create-dinosaur-route": [
+                Pattern("context.request.http.path", "matches", "/dinosaurs/?$"),
+                Pattern("context.request.http.method", "eq", "POST"),
+            ],
+            "metrics-federate-route": [
+                Pattern(path_fourth_element, "eq", "dinosaurs"),
+                Pattern("context.request.http.path", "matches", "/metrics/federate$"),
+            ],
+            "service-accounts-route": [Pattern(path_fourth_element, "eq", "service_accounts")],
+            "supported-instance-types-route": [Pattern(path_fourth_element, "eq", "instance_types")],
+            "agent-clusters-route": [Pattern(path_fourth_element, "eq", "agent-clusters")],
+            "admin-route": [Pattern(path_fourth_element, "eq", "admin")],
+            "acl-required": [
+                Pattern(path_fourth_element, "neq", "agent-clusters"),
+                Pattern(path_fourth_element, "neq", "admin"),
+            ],
+            "redhat-sso": [Pattern("auth.identity.iss", "eq", rhsso.well_known["issuer"])],
+            "admin-sso": [Pattern("auth.identity.iss", "eq", admin_rhsso.well_known["issuer"])],
+            "require-org-id": [Pattern("auth.identity.org_id", "neq", "")],
+        }
+    )
+    authorization.add_rule([PatternRef("api-route"), PatternRef("v1-route")])
+
+    authorization.identity.clear_all()
+    authorization.identity.add_oidc(
+        "redhat-sso",
+        rhsso.well_known["issuer"],
+        ttl=3600,
+        defaults_properties={"org_id": ValueFrom("auth.identity.middle_name")},
+    )
+    authorization.identity.add_oidc(
+        "admin-sso", admin_rhsso.well_known["issuer"], ttl=3600, when=[PatternRef("admin-route")]
```

Comment:
I believe the TTL was not tested for this exact use case, do you think it would make sense to add a test case for it also here?

Reply:
The new testcase can be implemented as part of #314. Added a new task to the issue.